### PR TITLE
improved error handling if something goes wrong

### PR DESCRIPTION
**Pull request by schiesbn**
Return error message in case something goes wrong while reading a file (e.g. because it is encrypted and the user doesn't has the correct key)

cc @PVince81 @nickvergessen  @th3fallen

### CODE REVIEW

File: ajax/loadfile.php [hunk 1]
```diff
@@ -30,25 +30,31 @@
...
-		'filecontents' => $filecontents,
-		'writeable' => $writeable,
-		'mime' => $mime,
-		'mtime' => $mtime))
-	);
-} else {
-	OCP\JSON::error(array('data' => array( 'message' => 'Invalid file path supplied.')));
+
+} catch (\Exception $e) {
+	OCP\JSON::error(array('data' => array('message' => $e->getMessage())));
```
Comment (DeepDiver1975) [inline ajax/loadfile.php]: see - exception messages sometimes hit the user :speak_no_evil:

**Comment by MorrisJobke** [comment]
Also this results in a broken interface. The dialog pops up, but then you can't click anything.

**Comment by MorrisJobke** [comment]
I guess the WebUI issues are fixed once #59 is merged. This PR just fixes the broken response from the server :)

File: ajax/loadfile.php (hunk 1 above)
Comment (nickvergessen) [inline ajax/loadfile.php]: We need to translate this? :secret:

Comment (schiesbn) [inline ajax/loadfile.php]: > see - exception messages sometimes hit the user

Yes, this is what we discussed when we worked in encryption 2.0. Do you remember? We decided to remove all the hard-coded encryption errors outside of the encryption app and agreed that instead the encryption app should throw useful errors which can then be catched and displayed to the user.

**Comment by schiesbn** [comment]
@MorrisJobke in order to test it you need this PR on core: https://github.com/owncloud/core/pull/16243

**Comment by schiesbn** [comment]
Error message is now translated if translation exists... Please review. Thanks!

File: ajax/loadfile.php [hunk 2]
```diff
@@ -30,25 +30,32 @@
...
+		}
+		$filecontents = iconv($encoding, "UTF-8", $filecontents);
+		OCP\JSON::success(array('data' => array(
+				'filecontents' => $filecontents,
+				'writeable' => $writeable,
+				'mime' => $mime,
+				'mtime' => $mtime))
+		);
+	} else {
+		OCP\JSON::error(array('data' => array('message' => 'Invalid file path supplied.')));
```
Comment (DeepDiver1975) [inline ajax/loadfile.php]: another (?) not translated message

Comment (schiesbn) [inline ajax/loadfile.php]: Didn't saw it... was already there before... But I will update it

Comment (schiesbn) [inline ajax/loadfile.php]: done :smile:

**Comment by DeepDiver1975** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**